lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -349,6 +349,17 @@ X86RegisterInfo::getCallPreservedMaskWithoutGPRs(const uint32_t *Mask) const {
   return CSR_NoRegs_RegMask;
 }
 
+const uint32_t *
+X86RegisterInfo::getCallPreservedMaskWithReturns(const uint32_t *Mask) const {
+  if (Mask == CSR_64_RegMask)
+    return CSR_64_WithRet_RegMask;
+  else if (Mask == CSR_Win64_RegMask)
+    return CSR_Win64_WithRet_RegMask;
+
+  // Be intentionally conservative.
+  return Mask;
```

Comment:
Sure, that seems better. I suppose that there's not really a reasonable default here.